zbus_macros/src/iface.rs

```diff
@@ -104,9 +108,27 @@ pub fn expand(args: AttributeArgs, mut input: ItemImpl) -> syn::Result<TokenStre
             }
         };
 
-    for method in &mut input.items {
-        let method = match method {
-            ImplItem::Method(m) => m,
+    for item in &mut input.items {
+        let (method, is_signal) = match item {
+            ImplItem::Fn(m) => (m, false),
+            // Since signals do not have a function body, they don't parse as ImplItemFn…
+            ImplItem::Verbatim(tokens) => {
+                // … thus parse them ourselves and construct an ImplItemFn from that
+                let decl = syn::parse2::<ImplItemSignal>(tokens.clone())?;
```

Comment:
This was the hardest piece to find out. Apparently syn 1.x just accepted `ImplItemMethod`s without a body, but syn 2.x does not. The error message when re-emitting this `ImplItem::Verbatim` was plenty confusing (it was "expected `impl`" on the first token of the item).